flickr.py

```diff
@@ -25,7 +25,7 @@ def params_unique_combination(baseurl, params_d, private_keys=["api_key"]):
             res.append("{}-{}".format(k, params_d[k]))
     return baseurl + "_".join(res)
 
-def search_flickr_by_tags(tags):
+def search_flickr(additional_params):
```

Comment:
This seems to be the right approach. A dictionary can be used to pass arbitrary number of key values and you can later merge it.